superset-frontend/src/dashboard/containers/DashboardPage.tsx

```diff
@@ -153,16 +154,47 @@ const DashboardPage: FC = () => {
   }, [readyToRender]);
 
   useEffect(() => {
```

Comment:
a little code style improvement. 
1. We don't have to catch exceptions here(catch exception logic in the API layer) 
2. We can make a utility function, e.g.:`isCachedFilterKey`, `isFilterObject`

```suggestion
  useEffect(() => {
    async function getDataMaskApplied(obj) {
      await getFilterValue(id, obj);
    }

    let dataMaskFromUrl;
    if (id > 0 && getUrlParam(URL_PARAMS.nativeFilters)) {
      dataMaskFromUrl = getUrlParam(URL_PARAMS.nativeFilters);
    }
    if (id > 0 && getUrlParam(URL_PARAMS.nativeFiltersKey)) {
      dataMaskFromUrl = getDataMaskApplied(getUrlParam(URL_PARAMS.nativeFiltersKey));
    }
    if (readyToRender) {
      if (!isDashboardHydrated.current) {
        isDashboardHydrated.current = true;
        if (isFeatureEnabled(FeatureFlag.DASHBOARD_NATIVE_FILTERS_SET)) {
          // only initialize filterset once
          dispatch(getFilterSets(id));
        }
      }
      dispatch(
        hydrateDashboard(
          dashboard,
          charts,
          filterboxMigrationState,
          dataMaskFromUrl,
        ),
      );
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [readyToRender, filterboxMigrationState]);
```